### ISSUE TITLE
StarSpace model

### ISSUE THREAD

**Issue by balinda-1995** [facebookresearch/StarSpace#167]
I am finding some difficulty understanding the starspace model after reading several time [the paper](https://arxiv.org/abs/1709.03856) 
I think I understood the training step: the goal is to learn the feature matrix and this is done by minimizing the loss function that has as input the similarity function (cosine) applied on entity pairs
What I did not understand is the test part. Suppose we have the matrix trained, if I enter a new document how can I use the model to classify it and find the right label, for example?

**Comment by ledw** [comment]
@balinda-1995 hi, in the test part, the model outputs the most likely label for a given document according to the similarity function. If you use the -predictionFile flag, then it will output the most likely top k results for each document. I hope that answers your question.

**Comment by balinda-1995** [comment]
@ledw Yes, Thanks for your answer!
